docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+A `Service` primarily consists of reusable lightweight stateless business logic components.
+Due to its deployment across all applications, a `Service` is constrained to accessing data providers that are available universally (for instance, the backend database is not accessible from storefront applications by default).
+
+```
+[Organization]
+└── Service
+    └── [Module]
+        ├── [CustomDirectory]
+        │   ├── [ModelName]Interface.php
+        │   └── [ModelName].php        
+        ├── [Module]Config.php            
+        ├── [Module]DependencyProvider.php
+        ├── [Module]ServiceFactory.php
+        ├── [Module]ServiceInterface.php
+        └── [Module]Service.php
+```
+
+Used components
+- [Config](#module-configurations)
+- [Dependency Provider](#dependency-provider)
+- [Factory](#factory)
+- [Mapper / Expander / Hydrator](#mapper--expander--hydrator)
+- [Model](#model)
+- [Service facade](#service-facade)
+
+## Shared
+
+The `Shared` library contains code that is designed to be utilized across different application layers and modules.
+The `Shared` library is intended to facilitate the sharing of code among various application layers and modules.
+To ensure compatibility and versatility across different application architecture setups, any content within the `Shared` library must be free of application layer-specific elements. Therefore, the use of [Factories](#factory) is not permitted within the `Shared` library.
+
+```
+[Organization]
+└── Shared
+    └── [Module]
+        ├── Transfer
+        │   └── [module_name].transfer.xml
+        ├── [Module]Constants.php
+        └── [Module]Config.php
+```
+
+Used components
+- [Config](#module-configurations)
+- [Constants](#module-configurations)
+- [Transfer](#transfer-object)
+
+# Layers
+
+An application layer can have up to four logical layers with clear purpose and communication rules.
+- **Presentation Layer**: contains twig-templates, JS and CSS files
+- **Communication Layer**: contains controllers, forms, tables and plugins
+- **Business Layer**: contains the business logic of a module
+- **Persistence Layer**: contains queries, entities and the schema of a module
+
+## Presentation Layer responsibilities
+- Responsible for handling the user interface (UI) presentation.
+- Contains frontend elements, such as HTML, Twig templates, JavaScript (JS), TypeScript, Cascading Style Sheets (CSS) files, and so on.
+- Handles user interactions and input validations on the client side.
+- Interacts with the [Communication Layer](#communication-layer-responsibilities) to retrieve necessary data for display.
+
+## Communication Layer responsibilities
+- Acts as an intermediary between the [Presentation Layer](#presentation-layer-responsibilities) and the [Business Layer](#business-layer-responsibilities).
+- Contains [controllers](#controller) responsible for handling HTTP requests and responses.
```

Comment:
```suggestion
- Contains console commands.
- Contains [controllers](#controller) responsible for handling HTTP requests and responses.
```

Reply:
It is mentioned bellow about plugins 
Console command is a plugin as well

Reply:
I have been working in Spryker for three years, and it is not clear for me that Console command is a plugin. We should explicitly mention that either here, or under the plugins.